src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -1020,6 +1024,10 @@ MapArchiveResult MetaspaceShared::map_archives(FileMapInfo* static_mapinfo, File
     assert(static_mapinfo->mapping_end_offset() == dynamic_mapinfo->mapping_base_offset(), "no gap");
   }
 
+  if (AutoCreateSharedArchive && static_mapinfo != NULL && dynamic_mapinfo == NULL) {
+    warning("AutoCreateSharedArchive will be ignored for static archive");
```

Comment:
To be consistent, we should use `log_warning(cds)`. Also, the message is unclear. Here, `foo.jsa` is a static archive:

```
$ java -XX:+AutoCreateSharedArchive -XX:SharedArchiveFile=foo.jsa -cp HelloWorld.jar HelloWorld
Java HotSpot(TM) 64-Bit Server VM warning: AutoCreateSharedArchive will be ignored for static archive
Hello World
```

I think it's better to print out:

```
AutoCreateSharedArchive is ignored because foo.jsa is a static archive
```